pkg/pgmodel/end_to_end_tests/concurrent_sql_test.go

```diff
@@ -108,3 +166,110 @@ func TestConcurrentSQL(t *testing.T) {
 		}
 	})
 }
+
+func testConcurrentInsertSimple(t testing.TB, db *pgxpool.Pool) {
+	metrics := []prompb.TimeSeries{
+		{
+			Labels: []prompb.Label{
+				{Name: MetricNameLabelName, Value: "cpu_usage"},
+			},
+			Samples: []prompb.Sample{
+				{Timestamp: 10, Value: 0.5},
+			},
+		},
+	}
+
+	ingestor := NewPgxIngestor(db)
+	defer ingestor.Close()
+	_, err := ingestor.Ingest(metrics)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func testConcurrentInsertAdvanced(t testing.TB, db *pgxpool.Pool) {
+	metrics := []prompb.TimeSeries{
+		{
+			Labels: []prompb.Label{
+				{Name: MetricNameLabelName, Value: "cpu_usage_a"},
+				{Name: "namespace", Value: "production"},
+				{Name: "node", Value: "brain"},
+			},
+			Samples: []prompb.Sample{
+				{Timestamp: 10, Value: 0.5},
+				{Timestamp: 40, Value: 0.6},
+			},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: MetricNameLabelName, Value: "cpu_usage_a"},
+				{Name: "namespace", Value: "dev"},
+				{Name: "node", Value: "pinky"},
+			},
+			Samples: []prompb.Sample{
+				{Timestamp: 10, Value: 0.1},
+				{Timestamp: 45, Value: 0.2},
+			},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: MetricNameLabelName, Value: "cpu_total_a"},
+				{Name: "namespace", Value: "production"},
+				{Name: "node", Value: "brain"},
+			},
+			Samples: []prompb.Sample{
+				{Timestamp: 10, Value: 0.5},
+				{Timestamp: 40, Value: 0.6},
+			},
+		},
+		{
+			Labels: []prompb.Label{
+				{Name: MetricNameLabelName, Value: "cpu_total_a"},
+				{Name: "namespace", Value: "dev"},
+				{Name: "node", Value: "pinky"},
+			},
+			Samples: []prompb.Sample{
+				{Timestamp: 10, Value: 0.1},
+				{Timestamp: 45, Value: 0.2},
+			},
+		},
+	}
+
+	ingestor := NewPgxIngestor(db)
+	defer ingestor.Close()
+	_, err := ingestor.Ingest(metrics)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestConcurrentInsert(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	withDB(t, *testDatabase, func(db *pgxpool.Pool, t testing.TB) {
+		var wg sync.WaitGroup
+		wg.Add(2)
```

Comment:
Very minor nit, usually you add inside the goroutine so its more understandable who you are waiting for :)

Reply:
I'd have to change it all over, so separate pr :)